README.md

```diff
@@ -4,23 +4,66 @@
 
 ## Table of Contents
 
-- [About](#about)
-- [Design](#design)
-- [Prerequisites](#prerequisites)
-- [Supported Kubernetes versions](#supported-kubernetes-versions)
-- [Setup](#setup)
-- [Automated testing](#automated-testing)
-- [Communication and contribution](#communication-and-contribution)
+- [Intel PMEM-CSI plugin for Kubernetes](#intel-pmem-csi-plugin-for-kubernetes)
+    - [Table of Contents](#table-of-contents)
+    - [About](#about)
+    - [Design](#design)
+        - [Architecture and Operation](#architecture-and-operation)
+        - [DeviceMode:LVM](#devicemodelvm)
+            - [Namespace modes in DeviceMode:LVM](#namespace-modes-in-devicemodelvm)
+            - [Using limited amount of total space in DeviceMode:LVM](#using-limited-amount-of-total-space-in-devicemodelvm)
+        - [DeviceMode:Direct](#devicemodedirect)
+            - [Namespace modes in DeviceMode:Direct](#namespace-modes-in-devicemodedirect)
+            - [Using limited amount of total space in DeviceMode:Direct](#using-limited-amount-of-total-space-in-devicemodedirect)
+        - [Driver modes](#driver-modes)
+        - [Driver Components](#driver-components)
+            - [Identity Server](#identity-server)
+            - [Node Registry Server](#node-registry-server)
+            - [Master Controller Server](#master-controller-server)
+            - [Node Controller Server](#node-controller-server)
+            - [Node Server](#node-server)
+        - [Communication channels](#communication-channels)
```

Comment:
I feel like "Communication channels" sounds incorrect, shall we rename this to something like "Driver communication".

Reply:
Fine with me.